CLONE_RESTORE.md

```diff
@@ -0,0 +1,284 @@
+
+1. **Problem:** The `s3` path wasn't correct, majorly because the tools within the postgres pod (WALE, WALG, PATRONI) were outdated.
+
+   **Solution:** Update the spilo image version used by the postgres operator to a more recent version and that should fix the issue.
+
+
+2. **Problem:** `FATAL:  no pg_hba.conf entry for host "[local]", user "postgres", database "postgres", SSL off`. This was caused because the Harbor Operator hardcoded the [pg_hba.config](https://github.com/goharbor/harbor-operator/blob/master/pkg/cluster/controllers/database/generate.go#L94) and it doesn't include the `postgres` user access to make replications.
+
+   **Solution:** `This is a possible solution that needs to be pushed upstream`.
+
+    To resolve this, edit the postgres operator CR and included the following under the patroni:pg_hba section and delete the pods.
+
+    **NOTE: This is not a sustainable solution as it requires a manual intervention anytime a harbor cluster is created. A [PR](https://github.com/goharbor/harbor-operator/pull/979) has been created to resolve this issue.**
+
+    ```yaml
+    pg_hba:
+    - hostssl all             all        0.0.0.0/0 md5
+    - host    all             all        0.0.0.0/0 md5
+    - local   all             all        trust
+    - local   replication     postgres   trust
+    - hostssl replication     postgres   all       md5
+    - local   replication     standby    trust
+    - hostssl replication     standby    all       md5
+    - hostssl all             +zalandos  all       pam
+    - hostssl all             all        all       md5
+    ```
+
+3. **Problem:** The `clone_with_wale.py` script doesn't use the correct `s3` path to download the backups for restoration.
+
+   **Solution:** The `CLONE_AWS_ENDPOINT` environment variable affects the `s3` path so remove it if you have it.
+
+
+### Post Recovery Options
+
+After a database has been recovered successfully from a backup from s3, the CLONE parameters can be removed from the `pod-config` configmap or left alone, this is because the postgresql service will always pick the latest data available regardless if the CLONE parameters are specified or not.
+
+## Summary
+
+There exists a solution for backup/restore of postgres. The solution is slightly less awkward for deployment solutions utilising the [zalando postgres operator](https://github.com/zalando/postgres-operator) directly, however it involves more operational overheard since the database is then decoupled from Harbor. The _issues_ outlined above could be resolved to remove friction of using the internal postgres operator (_which still is the zalando postgres operator, but it's configuration isn't as easily configurable_).
+
+Some important notes to consider:
+- Backups stored within the Postgres Backup S3 Bucket should be verified with backups listed by the `wal-g` cli within a Production environment setting (_alerting when the condition fails_).
```

Comment:
How to do that?

Reply:
I'd suggest the creation of an backup/restore system test when a repository containing the values associated with the `harbor-operator-app` and configuration values for `harbor-cluster` are known.